query.py

```diff
@@ -23,6 +23,19 @@ def _format_arg(self, arg_name, arg_val):
         '''
         return '{}:{}'.format(arg_name, arg_val)
 
+    def _query(self, http, url):
```

Comment:
There are two error return paths here:
`if res.status_code != requests.codes.ok` and `if 'errors' in json`

The error analysis in this function is OK, but what about the client of it? How would she know that there was an error detected? 
In all cases, error or not, you return the res object. And even though you know there was an error you still return the res object. That means that the client would also have to delve into the details of the response and figure out what had happened. 

Do you see this as a (big enough) problem?